fossunited/fossunited/web_template/recent_event_grants/recent_event_grants.html

```diff
@@ -175,6 +161,97 @@ <h6>{{grant.grant_amount}}</h6>
+            args: {
+                start: currentCount,
+                limit: limit
+            },
+            callback: function(r) {
+                if (!r.exc) { // Check for any exceptions
+                    if (r.message && r.message.grants) {
+                        const data = r.message;
+                        const container = document.getElementById('event-grant-category');
+                        data.grants.forEach(grant => {
+                            const grantElement = document.createElement('div');
+                            grantElement.className = 'grant-card';
+                            grantElement.innerHTML = formatGrantHTML(grant);
+                            container.appendChild(grantElement);
+                        });
+                        currentCount += data.grants.length; // Update count
+                        if (data.grants.length < limit) {
+                            document.getElementById('view-more').style.display = 'none'; // Hide button if no more grants
+                            document.getElementById('view-less').style.display = 'block';
+                        }
+
+                    } else {
+                        console.error("No grants data found in response");
+                    }
+
+                }
+            },
+            error: function(xhr, status, error) {
+                console.error('Error loading more grants:', status, error);
+            }
+        });
+    }
+
+    function viewLessGrants() {
+        // Remove all grant cards
+        const container = document.getElementById('event-grant-category');
+        while (container.firstChild) {
+            container.removeChild(container.firstChild);
+        }
+
+        // Reset currentCount and load the first 8 grants again
+        currentCount = 0;
+        limit = 8;
+        loadMoreGrants();
+
+        // Hide the view less button and ensure view more is visible
+        document.getElementById('view-less').style.display = 'none';
+        document.getElementById('view-more').style.display = 'block';
+    }
+
+    function formatGrantHTML(grant) {
+        return `
+            <h6>${grant.event_name}</h6>
+
+            <div class="event-detail" id="event-date">
+                <p>Date</p>
+                <p class="grant-card-value">${grant.event_start_date ? new Date(grant.event_start_date).toLocaleDateString('en-US', { day: '2-digit', month: 'short', year: 'numeric' }) : "TBA"}</p>
+            </div>
+
+            <div class="event-detail" id="event-location">
+                <p>Location</p>
+                <p class="grant-card-value">${grant.event_location || "TBA"}</p>
+            </div>
+
+            <div class="grant-card-footer">
+                <div class="event-detail">
+                    <p>Grant Amount</p>
+                    <p class="grant-card-value" id="grant-amount">${grant.grant_amount !== "Custom" ? grant.grant_amount : grant.custom_amount}</p>
```

Comment:
Where is this function used and what does this do?

Reply:
It's used in the [`loadMoreGrants()`](https://arc.net/l/quote/llbswnmu) and is used to make the fetched grant data formatted as cards and display.